interface/src/app/analyze/analysis/analysis.js

```diff
@@ -187,21 +221,20 @@ AnnotationType, SampleBin) {
     ]
   };
 
-  // TODO these exampleCols are temporarily hard-coded until a column chooser
-  // feature can be added
-  $scope.analysis = {
-    exampleCols: [
-      {'typename': 'genotype'},
-      {'typename': 'medium'},
-      {'typename': 'strain'}
-    ]
-  };
-
   // populate sample details
   // FIXME implement this loop as a method on SampleBin? (prob. with caching)
+  $scope.analysis.status = 'Retrieving sample details';
+  var pArrSamples = [];
   for (var i = 0; i < SampleBin.heatmapData.samples.length; i++) {
-    SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i]);
+    pArrSamples.push(
+      SampleBin.getSampleDetails(SampleBin.heatmapData.samples[i])
+    );
   }
+  $q.all(pArrSamples).then(function() {
```

Comment:
I like your way of using `$q.all()` to handle the promises. On the other hand, do you think it will be worth the time to add a new API on the backend to retrieve all data of samples that are related to a certain node? Then the frontend will be simpler. (See my latest PR, which adds a new API to get the data of all samples that are related to a given experiment.)